shuler_lab_to_nwb/fiberphotometry_datainterface/datainterface.py

```diff
@@ -0,0 +1,122 @@
+from typing import Optional
+import numpy as np
+import pandas as pd
+from pynwb.file import NWBFile
+from neuroconv.utils import load_dict_from_file
+from neuroconv.basetemporalalignmentinterface import BaseTemporalAlignmentInterface
+from neuroconv.tools.nwb_helpers import make_or_load_nwbfile
+from neuroconv.utils import FilePathType, dict_deep_update
+
+from .tools import read_fp_file, add_photometry
+
+
+class ShulerFiberPhotometryInterface(BaseTemporalAlignmentInterface):
+    """Data interface for Shuler lab Fiber Photometry data."""
+
+    def __init__(
+        self,
+        photometry_file_path: FilePathType,
+        metadata_file_path: FilePathType,
+        verbose: bool = True,
+    ):
+        """
+        Interface for writing Shuler Fiber Photometry data to NWB.
+
+        Parameters
+        ----------
+        photometry_file_path : FilePathType
+            Path to the file (.csv) containing the photometry time series data.
+        metadata_file_path : FilePathType
+            Path to the file (.yaml) containing the metadata for the photometry data.
+        verbose : bool, default: True
+            controls verbosity. ``True`` by default.
+        """
+        self.verbose = verbose
+        super().__init__(
+            photometry_file_path=photometry_file_path,
+            metadata_file_path=metadata_file_path
+        )
+    
+    def get_metadata_schema(self) -> dict:
+        metadata_schema = super().get_metadata_schema()
+        # metadata_schema["properties"]["FiberPhotometry"] = {
+        #     "type": "object",
+        #     "properties": {
+        #         "area": {"type": "string"},
+        #         "reference_max": {"type": "number"},
+        #         "signal_max": {"type": "number"},
+        #         "signal_reference_corr": {"type": "number"},
+        #         "snr": {"type": "number"},
+        #     },
+        # }
+        return metadata_schema
+    
+    def get_metadata(self) -> dict:
+        base_metadata = super().get_metadata()
+        metadata = load_dict_from_file(self.source_data["metadata_file_path"])
+        metadata = dict_deep_update(
+            d=base_metadata, 
+            u=metadata,
+            append_list=False,
+            compare_key="description"
+        )
+        return metadata
+    
+    def add_to_nwbfile(self, nwbfile: NWBFile, **conversion_options):
+        pass
+    
+    def get_original_timestamps(self) -> np.ndarray:
```

Comment:
Feel free to inherit from the `BaseDataInterface` now, you don't need to specify any of these methods then

Reply:
we'll probably need time sync for this interface when they share the behavioral data, that's why I'm using this class

Reply:
👍 do they know the TTL source and such? or is it externally parsed/stored vectors of timestamps?

Reply:
it is externally parsed/stored vectors of timestamps